spec/async_producer_spec.rb

```diff
@@ -67,4 +69,26 @@ def instrument(name, payload = {})
       expect(metric.payload[:message_count]).to eq 42
     end
   end
+
+  describe "#produce" do
+    it "delivers buffered messages" do
```

Comment:
Should we have a test case to cover the retrying ability?

Reply:
the second spec covers the retrying ability